src/avram/query.cr

```diff
@@ -1,11 +1,26 @@
 require "./queryable"
 
-abstract class Avram::Query
-  abstract def database : Avram::Database.class
-
+abstract class Avram::Query(T)
   # runs a SQL `TRUNCATE` on the current table
   def self.truncate
-    new.database.exec "TRUNCATE TABLE #{@@table_name}"
+    query = new
```

Comment:
I'm curious why assign an variable here?

Reply:
table_name is an instance method. This is the only place that needs it outside of instance methods so I didn't want to add table_name as a class method as well.